ganga/GangaCore/Lib/Localhost/LocalHostExec_batch.py.template

```diff
@@ -37,17 +37,18 @@ sys.stdout=open(join(outdir,'__syslog__'),'w')
 sys.stderr=sys.stdout
 
 print(f'Running with a batch size of {batch_size}')
-try:
-    pool = multiprocessing.Pool(batch_size)
-    pool.map(mp_worker, scriptnames_g)
-finally:
-    pool.close()
-    pool.join()
+if __name__ == '__main__':
+    try:
+        pool = multiprocessing.Pool(batch_size)
+        pool.map(mp_worker, scriptnames_g)
+    finally:
+        pool.close()
+        pool.join()
```

Comment:
This looks odd and makes me think that there is a different problem. If this if statement is ever False the code will do nothing. That can't be the solution. Is there a problem with the shebang at the start instead?

Reply:
@kh296 Can you have a look at which python interpreter the shebang actually starts for you. I suspect some mixed python environment to be the cause.

Reply:
I'm running in a `conda` environment, created from an `environment.yml` file containing:

```
name: test
dependencies:
  - pip
  - python
  - pip:
    - -e git+https://github.com/kh296/ganga.git@MacOS_no_htcondor#egg=ganga
```

I've checked that both ganga and the job script are using the environment's installation of python (3.10.0).

Reply:
OK, so the shebang is not actually used when running this inside Ganga. As you see from the [run](https://github.com/ganga-devs/ganga/blob/5bd0d25f5d99597201db4da51e907838761c2589/ganga/GangaCore/Lib/Localhost/Localhost.py#L133) method in the local backend, it is started directly as a `python3` process. This should still evaluate to the same though. It might be that some tweak to how the process is started will fix things though. You can try to change that line to
```python
            import sys
            process = subprocess.Popen([sys.executable, scriptpath, 'subprocess'], stdin=subprocess.DEVNULL, start_new_session=True)
```
and see if that is fixing the problems and allow you to run with the template unchanged.

Reply:
I've tried running with [LocalHostExec_batch.py.template](https://github.com/ganga-devs/ganga/blob/MacOS_compatibility/ganga/GangaCore/Lib/Localhost/LocalHostExec_batch.py.template) unchanged, and with revised [Localhost.run()](https://github.com/ganga-devs/ganga/blob/5fe8339c7bc7b854845bc94f647c65dffcff0b1b/ganga/GangaCore/Lib/Localhost/Localhost.py#L132).  Unfortunately we return to the previous situation on MacOS: the job script seems to get stuck in an infinite loop, continuously trying, and failing, to create the multiprocessing pool.